### PR TITLE
Remove controller support

### LINKED ISSUE #512: Remove controller support
It just is not feasible to use controllers for these game types, so remove all related code to using controllers as input.

- [ ] Remove in_joystick related code, files, commands, etc
- [ ] Remove controller settings in the base Options -> Mouse tab
- [ ] Remove all references to "Rumble" or vibration

### PR DESCRIPTION
**Pull request by Hat-Kid**
Closes #512 

This PR aims to remove as much of the joystick/controller and rumble implementation in Source as possible as we do not intend to support these features.

### Checklist
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

### CODE REVIEW

File: mp/src/game/client/view_effects.cpp [hunk 1]
```diff
@@ -394,7 +375,7 @@ screenshake_t *CViewEffects::FindLongestShake()
 //-----------------------------------------------------------------------------
 void CViewEffects::Shake( ScreenShake_t &data )
 {
-	if ( ( data.command == SHAKE_START || data.command == SHAKE_START_RUMBLEONLY ) && ( m_ShakeList.Count() < MAX_SHAKES ) )
+	if ( ( data.command == SHAKE_START ) && ( m_ShakeList.Count() < MAX_SHAKES ) )
```
Comment (Gocnak) [inline mp/src/game/client/view_effects.cpp]: Huh, I wonder if this was a bug on valve's part, shouldn't that second check be `== SHAKE_START_NORUMBLE`?

Comment (Hat-Kid) [reply mp/src/game/client/view_effects.cpp]: I was thinking that too, actually, should I change that?

File: mp/src/vgui2/vgui_controls/ControllerMap.cpp [hunk 2]
```diff
@@ -21,31 +21,6 @@ struct keystring_t
...
-												{ KEY_XBUTTON_RTRIGGER, 		"KEY_XBUTTON_RTRIGGER" },
-												{ KEY_XSTICK1_UP,				"KEY_XSTICK1_UP" },
-												{ KEY_XSTICK1_DOWN,				"KEY_XSTICK1_DOWN" },
-												{ KEY_XSTICK1_LEFT,				"KEY_XSTICK1_LEFT" },
-												{ KEY_XSTICK1_RIGHT,			"KEY_XSTICK1_RIGHT" },
-												{ KEY_XSTICK2_UP,				"KEY_XSTICK2_UP" },
-												{ KEY_XSTICK2_DOWN,				"KEY_XSTICK2_DOWN" },
-												{ KEY_XSTICK2_LEFT,				"KEY_XSTICK2_LEFT" },
-												{ KEY_XSTICK2_RIGHT,			"KEY_XSTICK2_RIGHT" } };
-
```
[the remark this answers is not in the document]
Comment (Hat-Kid) [reply mp/src/vgui2/vgui_controls/ControllerMap.cpp]: I removed it and it didn't seem to break anything at first glace so I suppose it's fine to remove it?

File: mp/src/game/client/view_effects.cpp (hunk 1 above)
Comment (Gocnak) [reply mp/src/game/client/view_effects.cpp]: Yeah I would make it check both, this can be called with start and with start_norumble

**Pull request merged**